### PR TITLE
Fixes for install

### PR DESCRIPTION
**Pull request by jborean93**
Some fixes for getting the install integration tests to pass. There are 2 outstanding issues I came across that will need to be fixed still

* `collection list` currently fails when coming across a dir that doesn't have a `galaxy.yml` or `MANIFEST.json`.
  * devel behaviour outputs an error and just treats the version as `*`
  * current PR behaviour stop the execution and fails
* when selecting the server to download from when the collection exists on muiltiple the select that is actually selected should be the first one in the `server_list`
  * current tests at https://github.com/webknjaz/ansible/compare/features/71784--resolvelib-collections-dep-resolver...jborean93:galaxy-resolver?expand=1#diff-e8f1119c538b4892135a92da1e3f1c523f2b1cd7658d7ebd975b19761c475187R151 are commented out until this is fixed

### CODE REVIEW

File: lib/ansible/galaxy/api.py [hunk 1]
```diff
@@ -263,6 +263,9 @@ def __init__(self, galaxy, name, url, username=None, password=None, token=None,
 
         display.debug('Validate TLS certificates for %s: %s' % (self.api_server, self.validate_certs))
 
+    def __str__(self):
```
Comment (webknjaz) [inline lib/ansible/galaxy/api.py:274]: Looks like this should be a `__repr__()`

File: lib/ansible/galaxy/collection/concrete_artifact_manager.py [hunk 2]
```diff
@@ -510,7 +501,7 @@ def _get_meta_from_installed_dir(
         with open(b_manifest_json_path, 'rb') as manifest_fd:
             b_manifest_txt = manifest_fd.read()
     except OSError:
-        raise LookupError(
+        raise AnsibleError(
```
Comment (webknjaz) [inline lib/ansible/galaxy/collection/concrete_artifact_manager.py]: This shouldn't raise a base error, when I added it, I've set up the translations on the calling side.

**Comment by webknjaz** [comment]
> `collection list` currently fails when coming across a dir that doesn't have a `galaxy.yml` or `MANIFEST.json`.
> * devel behaviour outputs an error and just treats the version as `*`

@s-hertel attempted to fix this @ [`66c5c7d` (#349)](https://github.com/webknjaz/ansible/pull/349/commits/66c5c7dd1ed133fca69fa28d382806f40cce4078) but I wanted to discuss this on the call.

File: lib/ansible/galaxy/dependency_resolution/providers.py [hunk 3]
```diff
@@ -182,6 +181,17 @@ def is_satisfied_by(self, requirement, candidate):
         A boolean should be returned to indicate whether `candidate` is a
         viable solution to the requirement.
         """
+        # NOTE: Only allow pre-release candidates if we want pre-releases or
+        # the req ver was an exact match with the pre-release version.
```
Comment (webknjaz) [inline lib/ansible/galaxy/dependency_resolution/providers.py:184]: Do we want this for the initial direct install requests or transitive dependencies too?

File: lib/ansible/galaxy/collection/galaxy_api_proxy.py [hunk 4]
```diff
@@ -52,13 +52,24 @@ def get_collection_versions(self, requirement):
...
-        return set(
-            (version, api)
-            for api in api_lookup_order
-            for version in api.get_collection_versions(
-                requirement.namespace, requirement.name,
-            )
-        )
+        version_information = set()
+        for api in api_lookup_order:
+            # TODO: Should this try/except logic be in api.py?
```
Comment (webknjaz) [inline lib/ansible/galaxy/collection/galaxy_api_proxy.py]: Probably not: api.GalaxyAPI encapsulates a connection to one server/endpoint but there's really no abstraction for querying things across servers + this one also attempts to treat access to the local artifacts the same way w/o any network I/O, like a facade.

File: lib/ansible/galaxy/api.py [hunk 5]
```diff
@@ -263,6 +263,9 @@ def __init__(self, galaxy, name, url, username=None, password=None, token=None,
...
+        return 'GalaxyAPI({name!s})'.format(name=to_native(self.name))
```
Comment (webknjaz) [inline lib/ansible/galaxy/api.py]:
```suggestion
        return '{name!s} Galaxy server'.format(name=to_native(self.name))
```

**Comment by webknjaz** [comment]
@jborean93 try rebasing this

**Comment by jborean93** [comment]
Sorry for the delay, I've rebased and split the changes into individual commits to fix a specific test.

**Pull request closed without merging**